### PR TITLE
Clevertap and GPS improvements

### PR DESCRIPTION
**Pull request by ankit209**

### CODE REVIEW

**Review by pnagle — COMMENTED** [review]
in MainApplication.java:
1. What's need for AppLifeCycleHelper?
2. Has the notification of driving shifted here ? 

in TrackerActivity:
1. Why workoutService used here instead of location receiver(need some brief idea abou this)?

in Analytics.java:
1. One version by Anshul and one by Ankit, would this merge be peaceful without any issues?
2. Why is still mixpanle called in Ankit's code?

in BaseActivity.java
1. Resolved the conflicts need to check the merged code has data from all relevant commits

in LoginImpl.java:
1. Check for IL employees

in SpikeFilter.java:
1. It shows the funciton empty

in ShareFragment.java
1. line 144 Radiobutton used in ankit's code is throwing an exception (Check fabric)

Rest looks good !!

File: app/build.gradle [hunk 1]
```diff
@@ -146,7 +147,15 @@ dependencies {
     compile 'com.google.android.gms:play-services-analytics:8.3.0'
     compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
 
-   // compile "com.google.firebase:firebase-core:8.3.0"
+    dependencies {
```
Comment (anshulagarwal06) [inline app/build.gradle:150]: dependencies block not require.

plugin must be outside dependencies blcok

File: app/src/main/java/com/sharesmile/share/rfac/fragments/ShareFragment.java [hunk 2]
```diff
@@ -129,13 +134,17 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
+        mWorkoutData = arg.getParcelable(WORKOUT_DATA);
         mShowLogin = arg.getBoolean(BUNDLE_SHOW_LOGIN);
     }
 
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        RadioButton button = (RadioButton)mShareGroup.findViewById(R.id.fb);
+        button.setButtonDrawable(VectorDrawableCompat.create(getResources(), R.drawable.selector_fb_radio, null));
```
Comment (anshulagarwal06) [inline app/src/main/java/com/sharesmile/share/rfac/fragments/ShareFragment.java:146]: we need to do same thing for other's too ( whatsapp,twitter)

File: app/build.gradle (hunk 1 above)
Comment (ankit209) [reply app/build.gradle:150]: Rectified.

File: app/src/main/java/com/sharesmile/share/rfac/fragments/ShareFragment.java (hunk 2 above)
Comment (ankit209) [reply app/src/main/java/com/sharesmile/share/rfac/fragments/ShareFragment.java:146]: This code block is not required now, I have removed it.
We have moved to app compat library 23.4.0, and using 
static
    {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }
code block in all activities.
This fixes the vectorDrawable crash.

**Comment by ankit209** [comment]
@pnagle Lets discuss all your review comments over call.

**Pull request merged**